src/main/java/com/conveyal/datatools/manager/jobs/AutoDeployJob.java

```diff
@@ -0,0 +1,209 @@
+     */
+    public AutoDeployJob(Project project, Auth0UserProfile owner) {
+        super(owner, "Auto Deploy Feed", JobType.AUTO_DEPLOY_FEED_VERSION);
+        projects.add(project);
+    }
+
+    /**
+     * Auto deploy all projects.
+     */
+    public AutoDeployJob(Auth0UserProfile owner) {
+        super(owner, "Auto Deploy Feed", JobType.AUTO_DEPLOY_FEED_VERSION);
+        projects = Persistence.projects.getAll();
+    }
+
+    @Override
+    public void jobLogic() {
+        for (Project project : projects) {
+            Deployment deployment = Persistence.deployments.getById(project.pinnedDeploymentId);
+            // Define if project and deployment are candidates for auto deploy.
+            if (!project.autoDeploy ||
+                project.pinnedDeploymentId == null ||
+                deployment == null ||
+                lockedProjects.contains(project.id)) {
+                LOG.info("Project {} skipped for auto deployment as required criteria not met.", project.name);
+                continue;
+            }
+            try {
+                synchronized (lockedProjects) {
+                    if (!lockedProjects.contains(project.id)) {
+                        lockedProjects.add(project.id);
+                    } else {
+                        continue;
+                    }
+                }
+                LOG.info("Auto deploy task running for project {}", project.name);
+                if (!canAutoDeployProject(project, deployment)) {
+                    continue;
+                }
+
+                List<String> feedVersionIdsForDeployment = getFeedVersionsForDeployment(project, deployment);
+                if (feedVersionIdsForDeployment.isEmpty()) {
+                    LOG.info("Project {} skipped for auto deployment as no newer feed versions are available!",
+                        project.name);
+                    continue;
+                }
+                if (deployment.latest() == null) {
+                    String message = String.format("Deployment `%s` has never been deployed. Skipping auto-deploy for project `%s`.",
+                        deployment.name,
+                        project.name);
+                    LOG.warn(message);
+                    NotifyUsersForSubscriptionJob.createNotification(
+                        "project-updated",
+                        project.id,
+                        message
+                    );
+                    continue;
+                }
+                // Send deployment (with new feed version) to most recently used server.
+                String latestServerId = deployment.latest().serverId;
+                OtpServer server = Persistence.servers.getById(latestServerId);
+                if (server != null) {
+                    // Advance all feed versions in the deployment.
+                    deployment.feedVersionIds = new ArrayList<>();
+                    deployment.feedVersionIds.addAll(feedVersionIdsForDeployment);
```

Comment:
This overwrites the existing list of feed version IDs which is ok as long as the newly calculated ones are correct.

Reply:
Addressed in Evan's PR.